homeassistant/components/twitch/coordinator.py

```diff
@@ -0,0 +1,164 @@
+        self._options = options
+
+    async def _get_channel_data(
+        self,
+        user: TwitchUser,
+        channel: TwitchUser,
+    ) -> TwitchChannelData:
+        """Get channel data."""
+        self.logger.debug("Channel: %s", channel.display_name)
+
+        stream: Stream | None = await first(
+            self._client.get_streams(user_id=[channel.id])
+        )
+        subscription: UserSubscription | None = None
+        following: FollowedChannelsResult | None = None
+
+        if self._client.has_required_auth(AuthType.USER, OAUTH_SCOPES):
+            try:
+                subscription = await self._client.check_user_subscription(
+                    user_id=user.id,
+                    broadcaster_id=channel.id,
+                )
+            except TwitchResourceNotFound:
+                self.logger.debug("User is not subscribed to: %s", channel.display_name)
+            except TwitchAPIException as exc:
+                self.logger.error(
+                    "Error response on check_user_subscription for %s: %s",
+                    channel.display_name,
+                    exc,
+                )
+            following = await self._client.get_followed_channels(
+                user_id=user.id,
+                broadcaster_id=channel.id,
+            )
+
+        followers = (await self._client.get_channel_followers(channel.id)).total
+
+        return TwitchChannelData(
+            channel,
+            stream=stream,
+            followers=followers,
+            following=following,
+            subscription=subscription,
+        )
+
+    async def _async_get_data(self) -> dict[str, TwitchChannelData]:
+        """Get data from Twitch."""
+        entity_registry = er.async_get(self.hass)
+
+        user = await first(self._client.get_users())
+        assert user
+
+        channel_options: list[str] = self._options[CONF_CHANNELS]
+
+        channels: list[TwitchUser] = []
+        # Split channels into chunks of 100 to avoid hitting the rate limit
+        for chunk in chunk_list(channel_options, 100):
+            async for channel in self._client.get_users(logins=chunk):
+                # Check if the entity is disabled
+                if (
+                    entity := entity_registry.async_get(
+                        f"sensor.{channel.display_name.lower()}"
+                    )
+                ) is not None and entity.disabled:
+                    self.logger.debug(
+                        "Channel %s is disabled",
+                        channel.display_name,
+                    )
+                    continue
+                channels.append(channel)
+
+        self.logger.debug("Enabled channels: %s", len(channels))
+
+        data: dict[str, TwitchChannelData] = {}
+
+        # Get data for each channel asynchronously as tasks
+        for channel_data in await asyncio.gather(
+            *[self._get_channel_data(user, channel) for channel in channels]
```

Comment:
Please create a coordinator per channel so we can remove the whole disabled check above. The coordinator will not fetch the data when the entity is disabled.